File: sdk/python/src/protomcp/result.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class ToolResult:
    result: str = ""
    is_error: bool = False
    enable_tools: Optional[list[str]] = None
    disable_tools: Optional[list[str]] = None
    error_code: Optional[str] = None
    message: Optional[str] = None
    suggestion: Optional[str] = None
    retryable: bool = False

File: sdk/go/protomcp/result.go
package protomcp

type ToolResult struct {
	ResultText   string
	IsError      bool
	ErrorCode    string
	Message      string
	Suggestion   string
	Retryable    bool
	EnableTools  []string
	DisableTools []string
}

func Result(text string) ToolResult {
	return ToolResult{ResultText: text}
}

func ErrorResult(text, errorCode, suggestion string, retryable bool) ToolResult {
	return ToolResult{
		ResultText: text,
		IsError:    true,
		ErrorCode:  errorCode,
		Message:    text,
		Suggestion: suggestion,
		Retryable:  retryable,
	}
}
